CMakeLists.txt

```diff
@@ -34,6 +34,37 @@ endif()
 
 find_ros1_package(std_msgs REQUIRED)
 
+# find all known ROS1 message/service packages
+execute_process(
+    COMMAND "rosmsg" "list"
+    OUTPUT_VARIABLE rosmsg_output
+    ERROR_VARIABLE rosmsg_error
+)
+execute_process(
+    COMMAND "rossrv" "list"
+    OUTPUT_VARIABLE rossrv_output
+    ERROR_VARIABLE rossrv_error
+)
+if(NOT rosmsg_error STREQUAL "" OR NOT rossrv_error STREQUAL "")
+  message(FATAL_ERROR "${rosmsg_error}\n${rossrv_error}\nFailed to call rosmsg/rossrv\nPlease make sure to source your ROS2 workspace BEFORE sourcing your ROS1 workspace")
```

Comment:
The current recommendation is sourcing ROS 2 *after* ROS 1. That is to find CMake config files of ROS 2 packages over ROS 1 packages in case of the same package name. With this inverted order how is that supposed to work?